app/src/main/java/net/gini/android/gvlexample/gini/Extraction.java

```diff
@@ -0,0 +1,63 @@
+package net.gini.android.gvlexample.gini;
+
+import android.databinding.BaseObservable;
+import android.databinding.Bindable;
+import android.text.Editable;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+
+import net.gini.android.gvlexample.BR;
+
+/**
+ * Created by aszotyori on 23.11.17.
+ */
+
+public class Extraction extends BaseObservable {
+
+    private final String mName;
+    private String mValue;
+    private TextWatcher mValueWatcher = new TextWatcher() {
```

Comment:
Should this be part of the Extraction model?

Reply:
👍 Very good catch! It shouldn't, I should use an ObservableExtraction instead to add the data binding stuff to the model. I was hurrying to try out if data binding is viable. I'll clean it up.